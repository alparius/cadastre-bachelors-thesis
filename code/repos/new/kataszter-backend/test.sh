#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
