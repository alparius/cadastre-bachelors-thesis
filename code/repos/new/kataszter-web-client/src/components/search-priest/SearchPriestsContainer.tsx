import React from "react";
import { FormattedMessage as FM } from "react-intl";
import { Card, Container, Header, Pagination } from "semantic-ui-react";

import IPriestMin from "../../data/priest/PriestMin";
import CardPlaceholder from "../CardPlaceholder";
import ErrorMsg from "../ErrorMsg";
import PriestCard from "./PriestCard";

const PAGESIZE: number = 10;

interface ISearchPriestsContainerProps {
    priests: IPriestMin[] | undefined;
    loading: boolean;
    showError: boolean;
    currentPage: number;
    setCurrentPage(newPage: number): void;
}

const SearchPriestsContainer: React.FC<ISearchPriestsContainerProps> = (props: ISearchPriestsContainerProps) => {
    const { priests, loading, showError, currentPage, setCurrentPage } = props;

    const handlePageChange = (_: any, { activePage }: any) => setCurrentPage(activePage);

    if (showError) {
        return <ErrorMsg errorMessage={"error.connection"} />;
    } else if (loading) {
        return <CardPlaceholder />;
    } else if (!priests || priests.length === 0) {
        return (
            <>
                <CardPlaceholder />
                <Header as="h2">
                    <FM id="search.noResult" />
                </Header>
            </>
        );
    } else {
        return (
            <>
                {currentPage === 1 && (
                    <Header as="h2">
                        {priests.length} <FM id="search.size" />
                    </Header>
                )}
                <Card.Group stackable doubling itemsPerRow={2}>
                    {priests.slice((currentPage - 1) * PAGESIZE, currentPage * PAGESIZE).map((priest: IPriestMin) => (
                        <PriestCard key={priest.ID} priest={priest} />
                    ))}
                </Card.Group>
                <Container textAlign="center" style={{ marginTop: "3em" }}>
                    <Pagination activePage={currentPage} totalPages={Math.ceil(priests.length / PAGESIZE)} onPageChange={handlePageChange} />
                </Container>
            </>
        );
    }
};

export default SearchPriestsContainer;
