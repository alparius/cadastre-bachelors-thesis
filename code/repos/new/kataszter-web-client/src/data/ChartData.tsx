interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
