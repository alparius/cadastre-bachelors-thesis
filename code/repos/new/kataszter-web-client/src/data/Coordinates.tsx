interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
