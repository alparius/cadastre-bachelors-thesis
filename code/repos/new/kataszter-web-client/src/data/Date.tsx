interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
