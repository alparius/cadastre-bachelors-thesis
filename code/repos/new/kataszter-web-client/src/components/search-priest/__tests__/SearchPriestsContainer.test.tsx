/*===================
  SearchPriestsContainer.unit.test.tsx
 ===================*/
import React from "react";

import IPriestMin from "../../../data/priest/PriestMin";
import { NewTestPriestMin } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../util/test/test-wrapper";
import SearchPriestsContainer from "../SearchPriestsContainer";

const handlers = {
    setCurrentPage: jest.fn()
};

const priests: IPriestMin[] = [NewTestPriestMin];

describe("SearchPriestsContainer element", () => {
    afterEach(() => {
        handlers.setCurrentPage.mockClear();
    });

    it("should render a card group when the priests array is not empty", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchPriestsContainer priests={priests} loading={false} showError={false} currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card and a message when the priests array is empty", () => {
        const emptyPriests: IPriestMin[] = [];
        const container = createComponentWithIntlAndRouter(
            <SearchPriestsContainer priests={emptyPriests} loading={false} showError={false} currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError is true", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchPriestsContainer priests={priests} loading={false} showError currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an error message when showError and loading are both true", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchPriestsContainer priests={priests} loading showError currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a placeholder card when only loading is true", () => {
        const container = createComponentWithIntlAndRouter(
            <SearchPriestsContainer priests={priests} loading showError={false} currentPage={1} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });
});
