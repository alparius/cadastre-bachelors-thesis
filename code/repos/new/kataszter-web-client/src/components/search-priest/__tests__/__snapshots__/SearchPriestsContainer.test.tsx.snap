// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPriestsContainer element should render a card group when the priests array is not empty 1`] = `
Array [
  <h2
    className="ui header"
  >
    1
     
    Priests found matching your search
  </h2>,
  <div
    className="ui doubling stackable two cards"
  >
    <a
      aria-current={null}
      className="ui fluid card"
      href="/priests/profile/1"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          <h2
            className="ui header"
          >
            name
            <span
              style={
                Object {
                  "float": "right",
                }
              }
            >
              <i
                aria-hidden="true"
                className="grey question circle outline fitted icon"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              />
            </span>
          </h2>
        </div>
        <div
          className="meta"
        >
          born
          : 
          birthTown
          ,
           
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
           - 
          died
          : 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
        </div>
        <div
          className="meta"
        >
          subscribed
          : 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
          , 
          and
           
          graduated
          : 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
        </div>
        <div
          className="meta"
        >
          country
          : 
          birthCountry
          , 
          county
          : 
          birthCounty
          , 
          diocese
          : 
          diocese
        </div>
        <div
          className="description"
          style={
            Object {
              "marginTop": "1em",
            }
          }
        >
          <div
            className="ui list"
            role="list"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <div
                className="header"
              >
                Assistant priest placements
              </div>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <a
                aria-current={null}
                className="hoverLink"
                href="/parishes/profile/0"
                onClick={[Function]}
              >
                 
                place
                 
              </a>
              : 
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                2000.06.06
              </span>
               - 
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                2000.06.06
              </span>
            </div>
          </div>
          <div
            className="ui list"
            role="list"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <div
                className="header"
              >
                Main priest placements
              </div>
            </div>
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <a
                aria-current={null}
                className="hoverLink"
                href="/parishes/profile/0"
                onClick={[Function]}
              >
                 
                place
                 
              </a>
              : 
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                2000.06.06
              </span>
               - 
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                2000.06.06
              </span>
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>,
  <div
    className="ui center aligned container"
    style={
      Object {
        "marginTop": "3em",
      }
    }
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="First item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="firstItem"
        value={1}
      >
        «
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="prevItem"
        value={1}
      >
        ⟨
      </a>
      <a
        aria-current={true}
        aria-disabled={false}
        className="active item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="nextItem"
        value={1}
      >
        ⟩
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Last item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="lastItem"
        value={1}
      >
        »
      </a>
    </div>
  </div>,
]
`;

exports[`SearchPriestsContainer element should render a placeholder card and a message when the priests array is empty 1`] = `
Array [
  <div
    className="ui fluid card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="header"
          >
            <div
              className="line"
            />
          </div>
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="ui placeholder"
        >
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
        </div>
      </div>
    </div>
  </div>,
  <h2
    className="ui header"
  >
    No results were found for your search. Try widening the search criterias.
  </h2>,
]
`;

exports[`SearchPriestsContainer element should render a placeholder card when only loading is true 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="header"
        >
          <div
            className="line"
          />
        </div>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchPriestsContainer element should render an error message when showError and loading are both true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;

exports[`SearchPriestsContainer element should render an error message when showError is true 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "7em",
    }
  }
>
  <h2
    className="ui header red"
  >
    Could not estabilish server connection.
  </h2>
  <h3
    className="ui header"
  >
    <a
      aria-current="page"
      className="hoverLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to the Home Page
    </a>
  </h3>
</div>
`;
